crates/ty_python_semantic/src/module_name.rs

```diff
@@ -127,6 +127,56 @@ impl ModuleName {
         true
     }
 
+    /// Given a parent module name of this module name, return the relative
+    /// portion of this module name.
+    ///
+    /// For example, a parent module name of `importlib` with this module name
+    /// as `importlib.resources`, this returns `resources`.
+    ///
+    /// If `parent` isn't a parent name of this module name, then this returns
+    /// `None`.
+    ///
+    /// # Examples
+    ///
+    /// This example shows some cases where `parent` is an actual parent of the
+    /// module name:
+    ///
+    /// ```
+    /// use ty_python_semantic::ModuleName;
+    ///
+    /// let this = ModuleName::new_static("importlib.resources").unwrap();
+    /// let parent = ModuleName::new_static("importlib").unwrap();
+    /// assert_eq!(this.relative_to(&parent), ModuleName::new_static("resources"));
+    ///
+    /// let this = ModuleName::new_static("foo.bar.baz.quux").unwrap();
+    /// let parent = ModuleName::new_static("foo.bar").unwrap();
+    /// assert_eq!(this.relative_to(&parent), ModuleName::new_static("baz.quux"));
+    /// ```
+    ///
+    /// This shows some cases where it isn't a parent:
+    ///
+    /// ```
+    /// use ty_python_semantic::ModuleName;
+    ///
+    /// let this = ModuleName::new_static("importliblib.resources").unwrap();
+    /// let parent = ModuleName::new_static("importlib").unwrap();
+    /// assert_eq!(this.relative_to(&parent), None);
+    ///
+    /// let this = ModuleName::new_static("foo.bar.baz.quux").unwrap();
+    /// let parent = ModuleName::new_static("foo.barbaz").unwrap();
+    /// assert_eq!(this.relative_to(&parent), None);
+    /// ```
+    #[must_use]
+    pub fn relative_to(&self, parent: &ModuleName) -> Option<ModuleName> {
```

Comment:
Should/could we use this in https://github.com/astral-sh/ruff/blob/c9dff5c7d5bcab2a83f1c3b4a1a80af230ece541/crates/ty_python_semantic/src/module_name.rs#L291-L295

Reply:
I think that function is doing something different. A better name for that function might be `module_name_from_relative_import`

Reply:
Aside from using `relative_module_name` directly and taking this suggestion more holistically, I don't see a simpler way of utilizing `ModuleName::ancestors` (or `ModuleName::components`) to get what we want here.

Reply:
Oh. Derp. Yes.